src/OpenTelemetry.Instrumentation.AWSLambda/AWSLambdaWrapper.cs

```diff
@@ -186,20 +188,20 @@ internal static Activity OnFunctionStart<TInput>(TInput input, ILambdaContext co
 
         // We assume that functionTags and httpTags have no intersection.
         var activityName = AWSLambdaUtils.GetFunctionName(context) ?? "AWS Lambda Invoke";
-        var activity = AWSLambdaActivitySource.StartActivity(activityName, ActivityKind.Server, parentContext, functionTags.Concat(httpTags), links);
+        var activity = AWSLambdaActivitySource.StartActivity(activityName, ActivityKind.Server, parentContext, functionTags.Concat(httpTags)!, links);
```

Comment:
Or is this about inner nulls? In that case we need to check why we get them inferred, see also https://github.com/open-telemetry/opentelemetry-dotnet-contrib/pull/1295#discussion_r1287283098

Reply:
Ah, okay, so it's because the compiler needs to consider the case that StartActivity would write into the list of tags, and insert a null into a variable we annotated as not-null. The ! in this position could mean so many different things, I wonder if there would be any possibility to express this more cleanly, but I can't think of anything. But adding a comment here might be good.